lib/internal/tty.js

```diff
@@ -82,9 +82,9 @@ function warnOnDeactivatedColors(env) {
   if (warned)
     return;
   let name = '';
-  if (env.NODE_DISABLE_COLORS !== undefined)
```

Comment:
These entries could (IMO should) all be simplified by just checking for truthiness. An env may only be `undefined` or a string due to the internal setter, no matter that e.g., `null` is assigned. It would result in `'null'`.

Reply:
Thinking about it again: a user may pass through arbitrary values as environment variables. If a zero would for example be passed through, it would now be handled as not defined, while it would be picked up as process.env (being transformed to a string before).

So maybe we should just handle it as the stringified version? As in: `env.FOO?.toString()`? I am somewhat uncertain about how to handle this case ideally.

Reply:
I don't think it's a concern, IIUC it's always coerced to a string:

```console
$  node -e 'process.env.TEST=null;console.log(typeof process.env.TEST)'
string
$ node -e 'process.env.TEST=0;console.log(typeof process.env.TEST)'   
string
```

Reply:
I mean in case the API is used programatically by passing through the envs with an object and not relying on the default (that falls back to process.env)

Reply:
Yes, let's do that for now and let's add a test case for this.